crates/ruff_linter/src/renamer.rs

```diff
@@ -202,7 +210,21 @@ impl Renamer {
                 // Rename the references to the binding.
                 edits.extend(binding.references().map(|reference_id| {
                     let reference = semantic.reference(reference_id);
-                    Edit::range_replacement(target.to_string(), reference.range())
+                    let range_to_replace = {
+                        if reference.in_dunder_all_definition() {
+                            let reference_source = locator.slice(reference.range());
+                            let relative_range = raw_contents_range(reference_source)
```

Comment:
What happens here if you have something like:

```
__all__ = ["Se" 't'"]
```

I.e., mismatching quotes?

Reply:
Aha, we produce invalid syntax! Good catch...